Revamp Function BBEs (#3840)
ballerina-platform/ballerina-distribution: examples/function-closure/function_closure.md

# Function closure

> The object constructor and function can work as a closure, which captures its surrounding environment and can access variables outside of its own scope. This means that a closure can access variables that are defined in a different scope such as the global scope or the scope of a parent.

Reviewer: ```suggestion
The object constructor and function can work as a closure, which captures its surrounding environment and can access variables outside of its own scope. This means that a closure can access variables that are defined in its parent scopes.
```

> The object constructor and function can work as a closure, which captures its surrounding environment and can access variables outside of its own scope. This means that a closure can access variables that are defined in a different scope, such as the global scope or the scope of a parent.

Reviewer 2: ```suggestion
The object constructor and function can work as a closure, which captures its surrounding environment and can access variables outside of its own scope. This means that a closure can access variables that are defined in a different scope such as the global scope or the scope of a parent.
```

> It is a compile-time error to have parameter names same as outer-scope variable names.

Reviewer 2: ```suggestion
It is a compile-time error to have parameter names similar to the outer-scope variable names.
```
